SpellingBee/create_spelling_data.py

```diff
@@ -0,0 +1,320 @@
+    sims = get_cosine_sim(emb)
+    top_k = torch.topk(sims, k=k, dim=-1)[1]
+    return top_k[test_set, :]
+
+
+def is_whole_word(token_index, cur_dict, fname):
+    if 'roberta' in fname:
+        return 'Ġ' in cur_dict[token_index]
+    return None
+
+
+def get_similarity_split(fname, cur_dict, file_len, k, test_indices, whole_words_only, with_lemma=True, verbose=False):
+    ''' Returns a data split of random words in tests, and words that are not similar in train'''
+    emb_matrix = get_emb_matrix(fname)
+
+    test_normalized_tokens = set()
+    if with_lemma:
+        for token_index in test_indices:
+            test_normalized_tokens.add(normalize(token_index, cur_dict, fname))
+
+    most_similar_words_to_test_set = get_top_k_similar(emb_matrix, test_indices, k)
+
+    if verbose:
+        print("test_normalized_tokens", test_normalized_tokens)
+        print("most_similar_words_to_test_set", most_similar_words_to_test_set)
+    count_normal_in_test = 0
+    count_normal_in_test_and_not_similar = 0
+
+    train_indices = []
+    for i in range(file_len):
+        if i not in most_similar_words_to_test_set and normalize(i, cur_dict, fname) not in test_normalized_tokens:
+            if whole_words_only and not is_whole_word(i, cur_dict, fname):
+                continue
+            train_indices.append(i)
+        if verbose:
+            if normalize(i, cur_dict, fname) in test_normalized_tokens:
+                print("This word normalized version is in test:", end=' |')
+                print(cur_dict[i], end=' | normalize:')
+                print(normalize(i, cur_dict, fname))
+                count_normal_in_test += 1
+            if i not in most_similar_words_to_test_set and normalize(i, cur_dict, fname) in test_normalized_tokens:
+                print("Not 20 closest, but it's normalized version is in test:", end=' |')
+                print(i, end=' |')
+                print(cur_dict[i], end=' | normalize:')
+                print(normalize(i, cur_dict, fname))
+                count_normal_in_test_and_not_similar += 1
+            print("count_normal_in_test:", count_normal_in_test)
+            print("count_normal_in_test_and_not_similar:", count_normal_in_test_and_not_similar)
+
+    return np.random.permutation(train_indices)
+
+
+def get_file_len(fname):
+    with open(fname) as f:
+        for i, l in enumerate(f):
+            pass
+    return i + 1
+
+
+def get_random_split(fname, cur_dict, file_len, test_indices, whole_words_only):
+    train_indices = []
+    if whole_words_only:
+        indices_to_choose_from = [i for i in range(file_len) if is_whole_word(i, cur_dict, fname)]
+    else:
+        indices_to_choose_from = range(file_len)
+
+    for i in indices_to_choose_from:
+        if i not in test_indices:
+            train_indices.append(i)
+    return np.random.permutation(train_indices)
+
+
+def get_split_indicies(fname, cur_dict, file_len, split_state, train_size, whole_words_only, test_indices=None):
```

Comment:
Why do you always return `None` as the second value from this function?
Also please raise an error if no if passes